### PR TITLE
dashboard: Small cleanups to dashboard

### PR DESCRIPTION
**Pull request by bdraco**
# What does this implement/fix?

dashboard: Small cleanups to dashboard

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>

## Test Environment

- [ ] ESP32
- [ ] ESP32 IDF
- [ ] ESP8266
- [ ] RP2040
- [ ] BK72xx
- [ ] RTL87xx

## Example entry for `config.yaml`:

```yaml
# Example config.yaml

```

## Checklist:
  - [ ] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/dashboard/web_server.py [hunk 1]
```diff
@@ -504,30 +509,34 @@ def post(self):
...
         storage_json = StorageJSON.load(storage_path)
         if storage_json is None:
             self.send_error(404)
             return
 
         from esphome.components.esp32 import VARIANTS as ESP32_VARIANTS
-        from esphome.components.esp32 import get_download_types as esp32_types
-        from esphome.components.esp8266 import get_download_types as esp8266_types
-        from esphome.components.libretiny import get_download_types as libretiny_types
-        from esphome.components.rp2040 import get_download_types as rp2040_types
```
Comment (bdraco) [inline esphome/dashboard/web_server.py:518]: Its a bit faster to not import all of these when we know the one we need

File: esphome/dashboard/web_server.py [hunk 2]
```diff
@@ -698,29 +707,34 @@ async def get(self):
...
-        from esphome.components.rp2040.boards import BOARDS as RP2040_BOARDS
-        from esphome.components.rtl87xx.boards import BOARDS as RTL87XX_BOARDS
-
-        platform_to_boards = {
-            const.PLATFORM_ESP32: ESP32_BOARDS,
-            const.PLATFORM_ESP8266: ESP8266_BOARDS,
-            const.PLATFORM_RP2040: RP2040_BOARDS,
-            const.PLATFORM_BK72XX: BK72XX_BOARDS,
-            const.PLATFORM_RTL87XX: RTL87XX_BOARDS,
-        }
```
Comment (bdraco) [inline esphome/dashboard/web_server.py:714]: Its linear compares and only import the modules we need vs building a dict that we will throw away each time

**Review by jesserockz — APPROVED** [review]
Thanks @bdraco !!

**Comment by bdraco** [comment]
thanks!

**Pull request merged**